src/wallets/router/routes.component.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useEffect, useMemo, type PropsWithChildren } from "react";
+import { Switch, Route as Woute } from "wouter";
+import { Page } from "~components/page/page.component";
+import type {
+  CommonRouteProps,
+  RouteConfig
+} from "~wallets/router/router.types";
+import { useLocation } from "~wallets/router/router.utils";
+
+export interface RoutesProps {
+  routes: RouteConfig[];
+  diffLocation?: boolean;
+  pageComponent?: React.ComponentType<PropsWithChildren>;
+}
+
+// TODO: Consider adding `<AnimatePresence>` and `<BodyScroller>` inside here.
+
+// TODO: Consider adding a prop to `RouteConfig.parseParams` to parse
```

Comment:
Might do it later. Removes the need to parse params in the views, as parsing would be done at the router level.